src/cms/config/functions/slugValidator.js

```diff
@@ -34,6 +40,12 @@ module.exports = {
       throw strapi.errors.badRequest('Do not add slash to the end of slugs.');
     }
   },
+  // a trailing dash is not allowed
+  slugNoTrailingDashValidator: function (slug) {
+    if (slug?.slice(-1) === '-') {
+      throw strapi.errors.badRequest('Do not add dash to the end of slugs.');
+    }
```

Comment:
you could include a second check

if (slug.includes('-/')) {
      throw strapi.errors.badRequest('Do not add dash before a slash.');
}